nntrainer/tensor/tensor.h

```diff
@@ -1923,6 +1944,53 @@ class Tensor {
     return continuous;
   }
 
+  /**
+   * @brief Check if data is valid number
+   */
+  bool checkDataValidation(bool print = false) const {
+    bool ret = true;
+
+    // It only support FP16 and FP32
+    if (getDataType() != Tdatatype::FP32 && getDataType() != Tdatatype::FP16)
+      return true;
+
+    std::string data;
+    for (unsigned int i = 0, len = size(); i < len; ++i) {
+      if (print) {
+        if (getDataType() == Tdatatype::FP32) {
+          data.append(std::to_string(static_cast<float>(getData<float>()[i])) + ", ");
+        }
+#ifdef ENABLE_FP16
+        else if (getDataType() == Tdatatype::FP16) {
+          data.append(std::to_string(static_cast<float>(getData<_FP16>()[i])) + ", ");
+        }
+#endif
+      }
+
+      if (getDataType() == Tdatatype::FP32) {
+        if (std::isnan(getData<float>()[i]) | std::isinf(getData<float>()[i])) {
```

Comment:
```|``` ==> ```||```

Reply:
```||``` is often cheaper than ```|```

Use ```||``` or ```!std::isfinite()``` or ```std::fpclassify & (FP_NAN | FP_INFINITE)```

You may be in a big loop.

Reply:
It fixed to use isfinit()